server/manager/src/main/java/org/apache/accumulo/manager/tableOps/namespace/rename/RenameNamespace.java

```diff
@@ -63,19 +65,23 @@ public Repo<Manager> call(FateId fateId, Manager manager) throws Exception {
       Utils.checkNamespaceDoesNotExist(manager.getContext(), newName, namespaceId,
           TableOperation.RENAME);
 
-      final String tap = manager.getZooKeeperRoot() + Constants.ZNAMESPACES + "/" + namespaceId
-          + Constants.ZNAMESPACE_NAME;
+      final String tap = manager.getZooKeeperRoot() + Constants.ZNAMESPACES;
+
+      final String currentNamespaceName =
+          Namespaces.getNamespaceName(manager.getContext(), namespaceId);
+      final byte[] updatedNamespaceMap =
```

Comment:
The way this code works does not seem quite correct.  It is reading data form zookeeper and passing it into  mutate existing.  Need to let mutateExisting do all of the read and writing to zookeeper inorder to correctly handle concurrent updates.  Should probably do something similar to a previous comment I made about using mutateExisting.  May want to move this code that calls mutateExsting into the new NamespaceMapping class.

Reply:
If the check for the ID already existing is done in the mutateExisting, then the check to see if it exists already in line 65 could be removed in favor of the check inside the mutateExisting. However, if the goal is to preserve the same exceptions that are being thrown by `Utils.checkNamespaceDoesNotExist`, which I believe throws a checked exception, then some cleverness will need to be done to throw a custom unchecked (RTE) exception inside the mutateExisting lambda, and then outside, the mutateExisting, it would need to be converted back to the same checked exception that the `Utils` method would have thrown.

I think there's a few more places where the `Utils.checkNamespaceDoesNotExist` method is used that could also be replaced in this same way. Then, that utility method could just be deleted.

Reply:
Updated to correct use of mutateExisting. Will address checking already-existing namespace name in near future commit.